schemas/panels/time-series/mig.cuepart

```diff
@@ -0,0 +1,49 @@
+if #panel.type == "timeseries" || #panel.type == "graph" {
+    kind: "TimeSeriesChart"
+    spec: {
+        queries: [ for _, target in #panel.targets {
+            kind: "TimeSeriesQuery"
+            spec: {
+                plugin: {
+                    #target: target
+                    %(conditional_timeserie_queries)
+                }
+            }
+        }]
+        legend: {
+            position: #panel.options.legend.placement
+        }
+        // mapping table for the `unit` attribute
+        // TODO: factorize this for all panels?
+        #unitMapping: {
+            ms: "Milliseconds"
+            s: "Seconds"
+            m: "Minutes"
+            h: "Hours"
+            d: "Days"
+            percent: "Percent"
+            percentunit: "PercentDecimal"
+            bytes: "Bytes"
+            decbytes: "Bytes"
+        }
+        if #panel.fieldConfig.defaults.unit != _|_ if #unitMapping["\(#panel.fieldConfig.defaults.unit)"] != _|_ {
+            unit: {
+                kind: #unitMapping["\(#panel.fieldConfig.defaults.unit)"]
+            }
+        }
+        if #panel.fieldConfig.defaults.thresholds != _|_ {
+            thresholds: {
+                //default_color:                                                       // TODO how to fill this field?
```

Comment:
I'm not really sure how this would be known in the BE, but on the FE side default colors are inferred from the MUI theme of the app, I guess in BE we could hard-code a default color once we decide on a logo and color palette for the project?

Reply:
Actually nvm the equivalent Grafana property to what I called default_color would be [here](https://github.com/perses/perses/blob/main/ui/app/sample-data/node-exporter-full_rev23.json#L121), I didn’t like how they had the first item as null to define the default threshold color

Reply:
Was our [FE conversion code](https://github.com/perses/perses/tree/main/ui/app/sample-data/convert) helpful at all when you were building this out btw? Wondering what the best way for us to collaborate on this will be in the future, I’ll keep cue up to date as we add more options, but is there some way I can also specify the equivalent Grafana property?

Reply:
I must admit I didn't check it, I'll take a look tomorrow.
About your question, I added explanations in the [CUE documentation file](https://github.com/perses/perses/blob/antoinethebaud/grafana-migrate/docs/cue.md#migration-from-grafana), the entrypoint to access the grafana properties is either `#panel`, `#var` or `#target` depending on which kind of object is considered, e.g
```cue
legend: {
    position: #panel.options.legend.placement
}
```
Still, it's not always a simple remapping like this, e.g for the units as you might have seen we have to use an intermediary mapping table & some conditionals to avoid errors:
```cue
#unitMapping: {
    ms: "Milliseconds"
    s: "Seconds"
    m: "Minutes"
    h: "Hours"
    d: "Days"
    percent: "Percent"
    percentunit: "PercentDecimal"
    bytes: "Bytes"
    decbytes: "Bytes"
}
if #panel.fieldConfig.defaults.unit != _|_ if #unitMapping["\(#panel.fieldConfig.defaults.unit)"] != _|_ {
    unit: {
        kind: #unitMapping["\(#panel.fieldConfig.defaults.unit)"]
    }
}
```

Reply:
Oh I see, this is really cool! Yeah not sure how helpful some of the FE code would be, it was mostly focused on that older node exporter dashboard, but for ex the mapping for calculations looks similar: [convert-transformations](https://github.com/perses/perses/blob/main/ui/app/sample-data/convert/convert-transformations.ts)